unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/WalletHUD/WalletSectionHUDController.cs

```diff
@@ -0,0 +1,176 @@
+
+        private void OnCurrentPolygonManaBalanceChanged(double currentBalance, double _) =>
+            view.SetPolygonManaBalance(currentBalance);
+
+        private void OnWalletSectionVisible(bool isVisible, bool _)
+        {
+            if (ownUserProfile.isGuest)
+                return;
+
+            if (isVisible)
+            {
+                fetchEthereumManaCancellationToken = fetchEthereumManaCancellationToken.SafeRestart();
+                fetchPolygonManaCancellationToken = fetchPolygonManaCancellationToken.SafeRestart();
+                RequestEthereumManaAsync(fetchEthereumManaCancellationToken.Token).Forget();
+                RequestPolygonManaAsync(fetchPolygonManaCancellationToken.Token).Forget();
+            }
+            else
+            {
+                fetchEthereumManaCancellationToken.SafeCancelAndDispose();
+                fetchPolygonManaCancellationToken.SafeCancelAndDispose();
+            }
+        }
+
+        private async UniTask RequestEthereumManaAsync(CancellationToken cancellationToken)
+        {
+            while (true)
+            {
+                await UniTask.WaitUntil(() =>
+                    ownUserProfile != null && !string.IsNullOrEmpty(ownUserProfile.userId),
+                    cancellationToken: cancellationToken);
+
+                view.SetEthereumManaLoadingActive(true);
+                Promise<double> promise = theGraph.QueryEthereumMana(ownUserProfile.userId);
+                if (promise != null)
+                {
+                    await promise;
+                    dataStore.wallet.currentEthereumManaBalance.Set(promise.value);
+                    view.SetEthereumManaLoadingActive(false);
+                }
+
+                await UniTask.Delay(TimeSpan.FromSeconds(WalletUtils.FETCH_MANA_INTERVAL), cancellationToken: cancellationToken);
+
+                if (cancellationToken.IsCancellationRequested)
+                    break;
+            }
+        }
+
+        private async UniTask RequestPolygonManaAsync(CancellationToken cancellationToken)
+        {
+            while (true)
+            {
+                await UniTask.WaitUntil(() =>
+                    ownUserProfile != null && !string.IsNullOrEmpty(ownUserProfile.userId),
+                    cancellationToken: cancellationToken);
+
+                view.SetPolygonManaLoadingActive(true);
+                Promise<double> promise = theGraph.QueryPolygonMana(ownUserProfile.userId);
+                if (promise != null)
+                {
+                    await promise;
+                    dataStore.wallet.currentPolygonManaBalance.Set(promise.value);
+                    view.SetPolygonManaLoadingActive(false);
+                }
+
+                await UniTask.Delay(TimeSpan.FromSeconds(WalletUtils.FETCH_MANA_INTERVAL), cancellationToken: cancellationToken);
+
+                if (cancellationToken.IsCancellationRequested)
+                    break;
+            }
+        }
```

Comment:
Same case here, can we merge those 2 methods?

Reply:
Done!